TAScheduler/UnitTests/tests.py

```diff
@@ -76,71 +77,879 @@ class TestAdminCourseInstrAsgmt(TestCase):  # Kevin
+                    self.instrObj.assignInstrCourse(invalid_input)
 
 class TestInstructorGetInstrLecAsgmts(TestCase):  # Kiran
+    instrDB = None
+    lecDB = None
+    user = None  # for instructor
+    instrObj = None
+    adminObj = None  # for deleting course
+
+    def setUp(self):
+        self.user = User.objects.create(
+            email_address='TA@example.com',
+            password='TApassword',
+            first_name='TA',
+            last_name='User',
+            home_address='123 TA St',
+            phone_number='1234567890'
+        )
+        self.instrDB = Instructor.objects.create(user=self.user, max_assignments=2)  # max 2 assignment!
+        self.instrObj = InstructorObj(self.instrDB)
+        tempUserForAdmin = User(email_address="admin@example.com")  # HOPEFULLY OK W/O FIELDS?
+        tempAdmin = Administrator(user=tempUserForAdmin)
+        self.adminObj = AdminObj(tempAdmin)
+
+        # Course
+        self.courseDB = Course.objects.create(
+            course_id=101,
+            semester='Fall 2023',
+            name='Introduction to Testing',
+            description='A course about writing tests in Django.',
+            num_of_sections=3,
+            modality='Online',
+            credits=4)
+        # Section
+        self.sectionDB = Section.objects.create(
+            section_id=100 + 1,
+            course=self.courseDB,
+            location="location" + str(1),
+            meeting_time="mt" + str(1))
+        # Lecture
+        Lecture.objects.create(section_id=self.sectionDB)
+
+    # 1] 1 lecture assignment
+    def test_1Assignment(self):
+        self.instrObj.assignInstrCourse(self.lecDB)
+        self.assertEquals(self.instrObj.getInstrLecAsgmts(), 1, msg="should be 1 assigment")
+
+    # 2] 0 lab assignment
+    def test_0Assignment(self):
+        self.assertEquals(self.instrObj.getInstrLecAsgmts(), 0, msg="should be 0 assigments")
+
+    # 3] 1 lab "assignment" - not in db
+    def test_1AssignmentNoExistLab(self):
+        tempSection = Section(section_id=102)  # not "101", which exists already HOPEFULLY OK W/O ALL FIELDS?
+        tempLec = Lecture(section=tempSection)
+
+        self.instrObj.assignInstrCourse(tempLec)
+        self.assertEquals(self.instrObj.getInstrLecAsgmts(), 0, msg="shouldn't assign non-existing lecture")
+
+    # 4] 1->0 lec Assignment
+    def test_1to0Assignment(self):
+        self.instrObj.assignInstrCourse(self.lecDB)
+        self.adminObj.removeSection(self.lecDB)
+        self.assertEquals(self.instrObj.getInstrLecAsgmts(), 0, msg="added then removed lecture, should be 0")
+
+
+#don't think this is needed for our sprint 1?
+class TestInstructorLecTAAsmgt(TestCase): #is this an instructor assigning a TA to a lecture?
     pass
 
-
-class TestInstructorLecTAAsmgt(TestCase):
+#don't think this is needed for our sprint 1?
+class TestInstructorLabTAAsmgt(TestCase): #is this an instructor assigning a TA to a lab?
```

Comment:
You are probably correct on the first part. Also yes because Instructors are supposed be able to assign TAs who are assigned to the course to sections.